server/routerlicious/packages/services-core/src/clientManager.ts

```diff
@@ -40,5 +59,15 @@ export interface IClientManager {
      * Returns all clients currently connected including a keep alive time.
      * Should be used with delis read only client functionality.
      */
-    getTimedClients?(tenantId: string, documentId: string): Promise<Map<string, ITimedClient>>;
+    getSequencedClients?(tenantId: string, documentId: string): Promise<Map<string, ISequencedSignalClient>>;
```

Comment:
`getSequencedClients` and `extendSequencedClients` are both optional, so that's why I think we did not have to immediately update the [Redis ClientManager](https://github.com/microsoft/FluidFramework/blob/1fae8d00b74e220f93d0d396910b8795ecad535f/server/routerlicious/packages/services/src/redisClientManager.ts) (and other test implementations) to include these. If r11s or FRS want to enable this new set of features being added for read clients, I assume we would have to provide an implementation, correct?

Also, let's say we mistakenly enabled this new set of read client features (I think we had a setting for that in the other PR). Since `getSequencedClients` and `extendSequencedClients` are both optional, our build would not break if we do not provide an implementation. But how would the service behave in that case?

In other words, I'm just concerned that if we check this in without providing concrete implementations to `getSequencedClients` and `extendSequencedClients`, we might later forget about them when we enable the read client feature, and we won't have a build error blocking us to make us realize that issue.

Would you say it is a good idea to make `extendSequencedClients` and possibly `getSequencedClients` be mandatory, and include implementations of that as part of this PR? I can also help with that.

@pradeepvairamani @znewton can you also take a look at this and let me know if I'm missing something from our side? Thanks!

Reply:
Yes you would need to implement these in order to use this new read client stuff.

I could make the methods required and cause them to throw "Not Implemented" errors for now. That way you would be required to implement them when moving to this system since you would get errors when trying to use it. Would that work?
Since I don't use any of the RedisClientManager / test client manager code, I don't think I should be the one to implement those methods.

Reply:
That makes total sense. We will follow up later to provide the actual implementations in Redis ClientManager. For now, having the "Not Implemented" error thrown is already helpful as we will definitely see it happening if we enable the Deli read client feature.